.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,46 @@
+name: Build and push Docker image
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build-and-push:
+    name: Push Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+      - name: Docker Setup QEMU
+        uses: docker/setup-qemu-action@v1.0.1
+      - name: Docker Setup Buildx
+        uses: docker/setup-buildx-action@v1.0.3
+
+      - name: Docker Login
+        uses: docker/login-action@v1.6.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_DEFAULT_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-duration-seconds: 1200
+          role-session-name: "temp-session"
+          role-skip-session-tagging: true
+      - name: Amazon ECR Login
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build and push Docker images
+        uses: docker/build-push-action@v2.2.0
+        with:
+          tags: |
+            smartcontract/builder:1.1.0
+            ${{ secrets.AWS_ECR_ACCOUNT_URL }}/builder:1.1.0
+          platforms: linux/386,linux/amd64,linux/arm64
+          push: true
```

Comment:
I would make this a conditional where it's only true when the branch is master, and false when its not. That way we can change the workflow triggers to satisfy

Reply:
Hum, since this repo only publishes tags (not every commit to master) I decided to make the pipeline only publish upon branch and tags.